s/start.go

```diff
@@ -15,43 +15,66 @@ import (
 	"github.com/capatazlib/go-capataz/c"
 )
 
-// monitorLoop does the initialization of supervisor's children and then runs an
-// infinite loop that monitors each child error.
-func (sup Supervisor) monitorLoop(
+// runMonitorLoop does the initialization of supervisor's children and then runs
+// an infinite loop that monitors each child error.
+//
+// This function is used for both async and sync strategies, given this, we
+// receive an onStart and onTerminate callbacks that behave differently
+// depending on which strategy is used:
+//
+// 1) When called with the async strategy, these callbacks will interact with
+// gochans that communicate with the spawner goroutine.
```

Comment:
Looks like we return the error either way?

https://github.com/capatazlib/go-capataz/pull/25/files#diff-b78502991b2a94626fb3bd19df00f2e0R72-R74

Reply:
Yes, but the returned error is not handled when on an async call, you may notice in the code you linked, the error is sent to the `onTerminate` callback. That's the only way to track the error when spawning a goroutine (instead of running it in the current thread)